backend/Tim.Backend/Startup/ServiceExtensions.cs

```diff
@@ -137,85 +135,136 @@ public static void AddInstrumentationServices(this IServiceCollection services,
-                    ApplicationKey = secrets.KustoAppKey,
-                    ApplicationNameForTracing = "TIM-Query-To-Kusto",
-                };
-
-                return new KustoQueryClient(connectionString);
-            });
-
             var authenticationSchemes = new List<string>();
 
             services.AddAuthentication(JwtBearerDefaults.AuthenticationScheme)
                 .AddJwtBearer(options =>
                 {
+                    options.Audience = authConfigs.ClientId;
+                    options.Authority = $"https://login.microsoftonline.com/{authConfigs.ClientAuthority}";
                     options.TokenValidationParameters = new TokenValidationParameters
                     {
-                        ValidateIssuer = false,
-                        ValidateAudience = false,
+                        ValidateIssuer = true,
+                        ValidateAudience = true,
                         ValidateIssuerSigningKey = true,
-                        IssuerSigningKey = new SymmetricSecurityKey(Encoding.UTF8.GetBytes(authConfigs.SigningKey)),
+                        /*IssuerSigningKey = new SymmetricSecurityKey(Encoding.UTF8.GetBytes(authConfigs.SigningKey))*/
+                        ValidIssuer = "https://login.microsoftonline.com/{authConfigs.ClientAuthority}",
+                        ValidAudience = $"api://{authConfigs.ClientId}",
+                        RequireExpirationTime = true,
+                        ValidateLifetime = true,
+                        RequireSignedTokens = true,
                     };
+                    options.SaveToken = true;
                 });
 
+            services.AddSingleton(p =>
+            {
+                return ConfidentialClientApplicationBuilder.Create(authConfigs.ClientId)
+                    .WithAuthority($"https://login.microsoftonline.com/{authConfigs.ClientAuthority}")
+                    .WithClientSecret(authConfigs.ClientSecret)
+                    .Build();
+            });
+
             services.AddPolicyRegistry();
 
-            services.AddSingleton<IKustoQueryWorker, RunKustoQueryWorker>();
+            services.AddSingleton<ISharedCache, InMemoryCache>();
+        }
+
+        /// <summary>
+        /// Initialization code for the database client.
+        /// </summary>
+        /// <param name="host">The host object to modify.</param>
+        /// <exception cref="ArgumentNullException">Error if any expected configurations are null.</exception>
+        /// <returns>A <see cref="Task"/> representing the asynchronous operation.</returns>
+        public static async Task InitializeDatabaseAsync(this IWebHost host)
+        {
+            if (host == null)
+            {
+                throw new ArgumentNullException(nameof(host));
+            }
+
+            var scope = host.Services.CreateScope();
+            var dbService = scope.ServiceProvider.GetService<IDatabaseClient>();
+            await dbService.Connect();
```

Comment:
the way you have it set up now means the connection to the db is always there - is that something we want? in previous designs, for each transaction we connect to the db when we need vs having it be connected all the time....part of me is worried that this may unexpectedly cause the k8s pod to crash because of couchbase which I dont love

Reply:
afaik, keeping the connection open is considered best practice, whereas creating a new connection per transaction is consider unnecessary overhead. However, you make a good point in that it should be able to recover when the connection is lost, which currently it wouldn't.

Reply:
oh nice, couchbase has retrying strategies built-in, so when the connection goes down, it is able to restore it and retry.